test/explorer/data_frame_test.exs

```diff
@@ -699,4 +699,69 @@ defmodule Explorer.DataFrameTest do
     assert Enum.to_list(columns["x"]) == [1, 2, 3]
     assert Enum.to_list(columns["y"]) == ["a", "b", "c"]
   end
+
+  describe "lazy" do
```

Comment:
I suggest moving this to a separate module because we can easily be looking at a 2kLOC in the future. :)